### ISSUE TITLE
It is possible to create multiple agents with the same name

### ISSUE THREAD

**Issue by ocaballeror** [opensistemas-hub/osbrain#219]
Users shouldn't be doing this, but this behavior shouldn't be allowed in the first place. A RuntimeError is raised when multiple agents are created with the same name, but I suggest we throw an error ourselves right when the second agent is registered with an existing name.

### Example:
```python
from osbrain import run_nameserver
from osbrain import run_agent

ns = run_nameserver()
run_agent('agent')
run_agent('agent')
run_agent('agent')

agent = ns.proxy('agent')
agent.log_info('Hello world!')

ns.shutdown()
```

**Comment by Peque** [comment]
To avoid race conditions this should be implemented in a try-except way. Make sure agent registering in the name server is done in a sequential thread-safe way.

Maybe take this opportunity to allow for running an agent without an alias. We could generate an UUID as an alias and return the proxy to it. Maybe this should be a separate issue...

**Comment by Peque** [comment]
@ocaballeror Yeah, that would not be safe. Somehow, we should handle the "already registered exception" in the `ns.register(self.name, url)` line.

We must make sure that, when an agent tries to register in the name server, the name server processes the registration requests sequentially. If the name already exists, then the name server should raise an error.

We should also probably rewrite the `ns.register(self.name, url)` section a bit and avoid the use of plain `except:` with no filtering.

**Comment by ocaballeror** [comment]
I just found out that Pyro4 has an option in it's nameserver to do it. Calling `ns.register` with the `safe=True` parameter will check if the name is available or raise an error otherwise. I'll have to investigate about how thread safe it is, but for now I think I'll go with their implementation of this rather than doing the checks myself.

**Comment by ocaballeror** [comment]
I didn't find anything about this in Pyro's documentation, but from reading the code I can confirm the registration on the nameserver is done safely using thread locks. I'm going with this solution and will post a PR soon.

**Comment by Peque** [comment]
@ocaballeror Perfect. :blush:

**Comment by Peque** [comment]
Fixed by #222.
